platform/Callback.h

```diff
@@ -652,8 +652,9 @@ class Callback<R()> {
     /** Call the attached function
      */
     R call() const {
-        MBED_ASSERT(_ops);
+        if (_ops)
         return _ops->call(this);
+		else return (R) 0;
```

Comment:
Ah good catch. The callback class should probably not be changed in the pr.

It is invalid to call a callback with no attached function. If this behaviour is needed, the callback can be checked at the call site like so:
``` cpp
Callback<void()> cb;
if (cb) {
    cb();
}
```

Or if the code relies on a default behaviour, the callback can be provided a noop function be default:
``` cpp
static void donothing() {}
Callback<void()> cb = callback(donothing);

if (something) {
    cb = callback(something_function);
}

cb();
```

This small function is actually very hot in mbed-os, and small changes may have surprising impacts. I would suggest a separate pr if this change is desirable.